### ISSUE TITLE
False positive UnusedPrivateClass when using class references

### ISSUE THREAD

**Issue by dkuc84** [arturbosch/detekt#1390]
```kotlin
package my.pkg

import org.junit.jupiter.api.Test
import kotlin.test.assertNotEquals

class UnusedPrivateClassTest {

    private data class SomeClass(val name: String)

    private data class AnotherClass(val id: Long)

    @Test
    fun `verify class is used`() {
        val instance = SomeClass(name = "test")
        assertNotEquals(AnotherClass::class.java.simpleName, instance::class.java.simpleName)
    }
}
```

## Expected Behavior
No issues reported as `AnotherClass` is clearly referenced.

## Observed Behavior
`UnusedPrivateClass` violation.

## Steps to Reproduce
See the code above

## Context
In our library we are using reflections to build objects and corresponding unit tests verify the correct objects get created. This leads to a lot of unnecessary suppressions - see https://github.com/ExpediaDotCom/graphql-kotlin/pull/91

## Your Environment
* Version of detekt used: RC12
* Link to your project (if it's a public repository): https://github.com/ExpediaDotCom/graphql-kotlin

**Comment by arturbosch** [comment]
Hey,
thanks for reporting this. I've got a fix for this :).

**Comment by dkuc84** [comment]
Thanks for the awesome library and quick fix!
